daemon/cdi.go

```diff
@@ -105,3 +114,53 @@ func (c *cdiHandler) getErrors() error {
 	}
 	return err.ErrorOrNil()
 }
+
+// listDevices uses the CDI cache to list all discovered CDI devices.
+// It conforms to the deviceDriver.ListDevices function signature.
+func (c *cdiHandler) listDevices(ctx context.Context, cfg *config.Config) (deviceListing, error) {
+	var out deviceListing
+
+	// Collect global errors from the CDI cache (e.g., issues with spec files themselves).
+	for specPath, specErrs := range c.registry.GetErrors() {
+		for _, err := range specErrs {
+			if errors.Is(err, os.ErrNotExist) {
+				continue
+			}
+			out.Warnings = append(out.Warnings, fmt.Sprintf("CDI: Error associated with spec file %s: %v", specPath, err))
+		}
+	}
+
+	qualifiedDeviceNames := c.registry.ListDevices()
+	if len(qualifiedDeviceNames) == 0 {
+		return out, nil
+	}
+
+	for _, qdn := range qualifiedDeviceNames {
+		device := c.registry.GetDevice(qdn)
+		if device == nil {
+			log.G(ctx).WithField("device", qdn).Warn("CDI: Cache.GetDevice() returned nil for a listed device, skipping.")
+			out.Warnings = append(out.Warnings, fmt.Sprintf("CDI: Device %s listed but not found by GetDevice()", qdn))
+			continue
+		}
+
+		cdiLibSpec := device.GetSpec()
+		if cdiLibSpec == nil {
+			log.G(ctx).WithField("device", qdn).Warn("CDI: Device.GetSpec() returned nil, skipping.")
+			out.Warnings = append(out.Warnings, fmt.Sprintf("CDI: Could not get spec for device %s", qdn))
+			continue
+		}
+
+		specPath := cdiLibSpec.GetPath()
+		specVendor := cdiLibSpec.GetVendor()
+		specClass := cdiLibSpec.GetClass()
+		kind := specVendor + "/" + specClass
+
+		out.Devices = append(out.Devices, system.DeviceInfo{
+			Kind:       kind,
+			Name:       qdn,
+			Attributes: map[string]string{"SpecFile": specPath},
```

Comment:
Is there a usecase or this Attributes field single path option? Otherwise I would omit it.

Reply:
Not sure! I removed it and left only `Name` - let's keep it simple initially, we can always extend that later.